roles/docker/tasks/main.yml

```diff
@@ -1,32 +1,19 @@
 ---
 # tasks file for docker
 
-# Avoid docker 1.9 when installing origin < 1.2 or OSE < 3.2 on RHEL/Centos and
-# See: https://bugzilla.redhat.com/show_bug.cgi?id=1304038
+- name: Downgrade docker if necessary
+  command: >
+    {{ ansible_pkg_mgr }} downgrade -y docker{{ '-' + docker_version if docker_version != '' else '' }}
+  when: not openshift.common.is_atomic and docker_downgrade | bool
```

Comment:
Expanding on previous comments, I would say that something like the following may work:

- Add `docker_version = ''` to roles/docker/defaults/main.yml

```
- name: Get current installed version if docker_version is specified
  command: "{{ repoquery_command }}  --installed --qf '%{version}' docker"
  when:  not openshift.common.is_atomic | bool and docker_version != ''
  register: docker_version_result
  changed_when: false

- name: Downgrade docker if necessary
  command: "{{ ansible_pkg_mgr }} downgrade -y docker-{{ docker_version }}"
  register: docker_downgrade_result
  when: not docker_version_result | skipped and docker_version_result.stdout | default('0.0') | version_compare(docker_version, 'gt')

- name: Install docker
  action: "{{ ansible_pkg_mgr }} name=docker{{ '-' + docker_version if docker_version != '' else '' }} state=present"
  when: not openshift.common.is_atomic | bool and not docker_downgrade_result | changed

- name: Remove deferred deletion for downgrades from 1.9
  command: >
    sed -i 's/--storage-opt dm.use_deferred_deletion=true//' /etc/sysconfig/docker-storage
  when: docker_downgrade_result | changed and docker_version_result.stdout | default('0.0') | version_compare('1.9', 'gt') and docker_version | default('9999') | version_compare('1.9', 'lt')
```